src/layers/logging.rs

```diff
@@ -86,6 +94,140 @@ impl Layer for LoggingLayer {
     }
 }
 
+/// `LoggingReader` is a wrapper of `BytesReader`, with logging functionality.
+pub struct LoggingReader {
+    scheme: Scheme,
+    path: String,
+    op: Operation,
+    have_read: usize,
+    to_read: Option<u64>,
+    inner: BytesReader,
+}
+
+impl LoggingReader {
+    pub(crate) fn make_reader(
```

Comment:
```suggestion
    fn new(
```

- This struct doesn't need to be initiated outside.
- `make_reader` is not beautiful.

Reply:
Using `wrap` is better I think

Reply:
I prefer using `new` to keep the same style as other structs.

Reply:
Name it to `new` then, to keep pace with others.